query/filter/filter.go

```diff
@@ -303,6 +316,12 @@ func buildValueMatcher(input jsoniter.RawMessage, field *schema.QueryableField)
 			return nil, nil, err
 		}
 		collation = value.NewCollationFrom(apiCollation)
+
+		if buildForSecondaryIndex && collation.IsCaseInsensitive() {
```

Comment:
So we won't be supporting case-insensitive queries on secondary indexes fields?

Reply:
We can support it in the in-memory filter part but not part of what we use to build the range keys. Or at least I can't think of a way of making that work.

Reply:
Sure, we can figure it out iteratively.